samples/apps/copilot-chat-app/webapi/Program.cs

```diff
@@ -1,256 +1,67 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 
 using System.Net;
-using System.Reflection;
-using System.Text.Json;
-using Microsoft.AspNetCore.Authentication;
-using Microsoft.AspNetCore.Authentication.JwtBearer;
-using Microsoft.Identity.Web;
-using Microsoft.SemanticKernel;
-using Microsoft.SemanticKernel.AI.Embeddings;
-using Microsoft.SemanticKernel.Connectors.Memory.Qdrant;
-using Microsoft.SemanticKernel.Memory;
-using Microsoft.SemanticKernel.SkillDefinition;
-using Microsoft.SemanticKernel.TemplateEngine;
-using SemanticKernel.Service.Auth;
-using SemanticKernel.Service.Config;
-using SemanticKernel.Service.Skills;
-using SemanticKernel.Service.Storage;
+using Microsoft.AspNetCore.Hosting.Server;
+using Microsoft.AspNetCore.Hosting.Server.Features;
+using Microsoft.AspNetCore.Server.Kestrel.Core;
+using Microsoft.Extensions.Options;
 
 namespace SemanticKernel.Service;
 
-public static class Program
+/// <summary>
+/// Copilot Chat Service
+/// </summary>
+public sealed class Program
 {
-    public static void Main(string[] args)
+    /// <summary>
+    /// Entry point
+    /// </summary>
+    /// <param name="args">Web application command-line arguments.</param>
+    public static async Task Main(string[] args)
     {
         WebApplicationBuilder builder = WebApplication.CreateBuilder(args);
 
-        builder.Host.ConfigureAppSettings();
-
-        // Set port to run on
-        var serverPortString = builder.Configuration.GetSection("ServicePort").Get<string>();
-        if (!int.TryParse(serverPortString, out int serverPort))
-        {
-            serverPort = Constants.DefaultServerPort;
-        }
-
-        builder.WebHost.UseUrls($"https://*:{serverPort}");
-
-        // Add services to the DI container
-        AddServices(builder.Services, builder.Configuration);
-
+        // Load in configuration settings from appsettings.json, user-secrets, key vaults, etc...
+        builder.Host.AddConfiguration();
+
+        // Add in configuration options and Semantic Kernel services.
+        builder.Services
+            .AddLogging()
+            .AddSingleton<ILogger>(sp => sp.GetRequiredService<ILogger<Program>>()) // some services require an untemplated ILogger
```

Comment:
If this line is not made redundant, you might consider making this guy an ILogger<Kernel> since it's only in the Kernel that untemplated ILogger's are used

Reply:
Default loggers require a type and I would rather not special case Logger create for this. This was the simplest and gives the same behavior.